pkg/ipcache/ipcache.go

```diff
@@ -159,7 +197,45 @@ func (ipc *IPCache) deleteLocked(IP string) {
 		if len(ipc.identityToIPCache[identity]) == 0 {
 			delete(ipc.identityToIPCache, identity)
 		}
+
+		// Remove a reference for the prefix length if this is a CIDR.
+		if _, cidr, err := net.ParseCIDR(IP); err == nil {
+			pl, bits := cidr.Mask.Size()
+			switch bits {
+			case net.IPv6len * 8:
+				unrefPrefixLength(ipc.v6PrefixLengths, pl)
+			case net.IPv4len * 8:
+				unrefPrefixLength(ipc.v4PrefixLengths, pl)
+			}
+		}
+	}
+}
+
+// ToBPFData renders the ipcache into the relevant set of CIDR prefixes for
+// the BPF datapath to use for lookup.
+func (ipc *IPCache) ToBPFData() (s6, s4 []int) {
```

Comment:
The ipcache should be locked here.

Reply:
Good spotting, I completely missed this.